app/routes.py

```diff
@@ -1,2 +1,271 @@
+                "is_complete": False if new_task.completed_at == None else True
+            }
+        }, 201
+
+
+@tasks_bp.route("/<task_id>", methods=["GET","PUT","DELETE"])
+def handle_task_id(task_id):
+    task = Task.query.get(task_id)
+    if task == None:
+        return ("", 404)
+
+    if request.method == "GET":
+        # task_dict = {
+        #     "task": {
+        #         "id": task.task_id,
+        #         "title": task.title,
+        #         "description": task.description,
+        #         "is_complete": False if task.completed_at == None else True,
+        #     }}
+        # # task_response = {"task":(task.to_dict())}
+        # if task.goal_id is not None:
+        #     # print(f"Goal ID: {task.goal_id}")
+        #     task_dict["goal_id"] = task.goal_id
+        
+        # print(f"Task Dict: {task_dict}")
+        # return task_dict, 200
+
+        response_body = {
+        "task": (task.to_dict())
+        }
+        return jsonify(response_body), 200
+
+    if request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+
+        db.session.commit()
+
+        return {
+            "task" : {
+                "id" : task.task_id,
+                "title" : task.title,
+                "description" : task.description,
+                "is_complete" : False if task.completed_at == None else True
+            }
+        }, 200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+
+        return jsonify({
+            "details": f'Task {task.task_id} "{task.title}" successfully deleted'
+            }), 200
+
+
+def slack_bot(title):
+    slack_post = (f"Someone just completed the task {title}")
+    url_slack ="https://slack.com/api/chat.postMessage"
+    authentication=os.environ.get("SLACK_TOKEN")
+    query_params={
+        "channel":'C02J08B9S0N',
+        "text":slack_post
+    }
+    header = {
+        "Authorization": authentication
+    }
+    message=requests.post(url_slack,params=query_params,headers=header)
+    # print("slack test")
+    return message.json()
+
+    
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def handle_task_mark_complete(task_id):
+    task = Task.query.get(task_id) 
+
+    if task is not None: 
+        task.completed_at = datetime.now()
+        slack_bot(task.title) # added a helper function 
+        return jsonify({
+            "task":{
+                "id":task.task_id,
+                "title":task.title,
+                "description":task.description,
+                "is_complete":bool(task.completed_at)
+            }
+        }),200
+
+    else: 
+        return (""), 404
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def handle_task_mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task is not None:
+        if request.method == "PATCH":
```

Comment:
good use of PATCH! Does it save the new value in the database, though? the tests might not account for that after it checks the response body